WinUIGallery/ControlPages/ScratchPadPage.xaml.cs

```diff
@@ -0,0 +1,575 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices.WindowsRuntime;
+using Windows.Foundation;
+using Windows.Foundation.Collections;
+using Microsoft.UI.Xaml;
+using Microsoft.UI.Xaml.Controls;
+using Microsoft.UI.Xaml.Controls.Primitives;
+using Microsoft.UI.Xaml.Data;
+using Microsoft.UI.Xaml.Input;
+using Microsoft.UI.Xaml.Media;
+using Microsoft.UI.Xaml.Navigation;
+using Windows.Media.Capture.Frames;
+using Windows.Media.Capture;
+using Microsoft.UI.Xaml.Media.Imaging;
+using Windows.Media.MediaProperties;
+using Windows.Storage.Streams;
+using System.ComponentModel;
+using Microsoft.UI.Input;
+using Microsoft.UI.Xaml.Markup;
+using System.Xml;
+using Windows.Storage;
+
+namespace WinUIGallery.ControlPages
+{
+    public sealed partial class ScratchPadPage : Page
+    {
+        public ScratchPadPage()
+        {
+            this.InitializeComponent();
+
+            var xamlStr = ReadScratchPadXAMLinLocalSettings();
+
+            // If there is no stored XAML, load the default.
+            if (xamlStr == null || xamlStr.Trim().Length == 0)
+            {
+                xamlStr = GetDefaultScratchXAML();
+            }
+
+            m_oldText = xamlStr;
+            textbox.TextDocument.SetText(Microsoft.UI.Text.TextSetOptions.None, m_oldText);
+            var formatter = new XamlTextFormatter(textbox);

```

Comment:
make formatter a member?

Reply:
That might be useful if in the future XamlTextFormatter supports incremental parsing and updates. Today it doesn't have that and I think doesn't really benefit from caching. Won'tFix.